demos/iot_config.h

```diff
@@ -24,6 +24,11 @@
 #ifndef IOT_CONFIG_H_
 #define IOT_CONFIG_H_
 
+/* Disable test builds by default. */
+#ifndef IOT_BUILD_TESTS
```

Comment:
This is a demo config file intended to be used by applications.  There should be no references to test-specific constants in this file.

Reply:
ok, I kind of agree. Then do you suggest I put it to iot_mqtt_internal.h?
We reference it in source file here: https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/libraries/standard/mqtt/src/iot_mqtt_subscription.c#L700

Reply:
Can you define via compiler argument in the test build script?

Reply:
That is already there https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/v4_beta/scripts/ci_test_build.sh

But when coverity analyzes the code, it doesn't see  IOT_BUILD_TESTS is defined, which violates MISRA 20.9